### ISSUE TITLE
Create environment for AutoLaunch components

### ISSUE THREAD

**Issue by daneah** [daneah/iterm-components#3]
It would open a lot of doors to enable long-running components to use third-party packages.

Long-running components need to be placed in the `AutoLaunch` folder. [The docs](https://iterm2.com/python-api/tutorial/running.html) mention that scripts can have their own full environment via `pyenv` if done correctly, but I wasn't able to create a directory structure that seemed to work.

**Comment by harrygallagher4** [comment]
Even if you put a script in `~/Library/Application Support/iTerm2/Scripts/AutoLaunch/` you still have to manually enable it by checking `scripts > AutoLaunch > script.py` in iTerm. After that, it should automatically launch when you restart iTerm.

**Comment by daneah** [comment]
@harrygallagher4 Yep, that works like a charm. This issue is about being able to have an environment _around_ a script that enables you to use packages installed from e.g. PyPI.

**Comment by harrygallagher4** [comment]
Whoops, didn't read the issue thoroughly enough 😅

**Comment by 0xd3e** [comment]
@daneah according to the [tutorial](https://www.iterm2.com/python-api/tutorial/index.html#creating-a-new-script) you can do that with the `Full Environment` option.

This creates a virtual environment where you can use other packages.

I've tried that yesterday. The structure looks like this.

```
> pwd
/Users/danieltrautmann/Library/Application Support/iTerm2/Scripts/AutoLaunch

> tree -L 3
.
└── status_bar
    ├── iterm2env
    │   ├── iterm2env-metadata.json
    │   ├── pyenv
    │   ├── shims
    │   └── versions
    ├── setup.cfg
    └── status_bar
        └── status_bar.py
```

Since I'm working on something similar you did, I'd be happy to help. However I'm not a Python expert.

**Comment by daneah** [comment]
@danieltrautmann Thanks for the additional info. I keep meaning to look (and respond to you!) but haven't been able to find the time just yet. I don't have a particular component in mind to test this out with yet, either, but maybe soon 😄 if you do in the meantime and want to propose any additions, I'd welcome them 💖

**Comment by 0xd3e** [comment]
I'll try to create a component that shows due tasks of a Todoist account.

**Comment by 0xd3e** [comment]
It seems that iTerm2 only installs the virtualenv when you create a new script via the GUI. So it is hard to provide scripts.

iTerm2 creates a file called `setup.cfg` and I thought this is used to setup the environment, but when you copy the directory containing the scripts to the iTerm2 `Scripts` directory. It doesn't create the environment automatically and I wasn't able to find a way to do it manually.

This is what the setup.cfg looks like:

```ini
# WARNING: This file was generated automatically by iTerm2. Do not edit it manually.
[metadata]
name=foo
version=1.0

[options]
scripts=foo/foo.py
install_requires=todoist-python; iterm2
python_requires = =3.8

[iterm2]
environment = >=62
```

Not sure if it is possible to use an environment which is not managed by iTerm2.

**Comment by daneah** [comment]
@danieltrautmann Thanks so much for taking the time to investigate. That's pretty interesting. This might be enough for me to explore further, soon. `setup.cfg` is a common way to provide installation details, and `install_requires` is what's probably important here. The syntax seems fine, so might just be a matter of triggering iTerm2 to act on it.

**Comment by 0xd3e** [comment]
Yes, the trigger is the part I couldn't find. Another option that came to my mind is to provide iterm-components as a pip package and let the user create a new script with iterm-components as dependency.

But then we would also need some kind of script to initialize iterm-components. In the end this is not an ideal solution.
